src/server/es/index.js

```diff
@@ -52,12 +61,86 @@ class ES {
       index: esIndex,
       type: esType,
       body: validatedQueryBody,
-    }).then(resp => resp, (err) => {
+    }).then(resp => resp.body, (err) => {
       log.error('[ES.query] error when query');
-      console.trace(err.message);
+      console.trace(err.message); // eslint-disable-line no-console
     });
   }
 
+  /**
+   * Fetch elastic search data using scroll API
+   * See https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-scroll.html
+   * @param {string} esIndex
+   * @param {string} esType
+   * @param {Object} argument - arg object for filter, fields, and sort
+   */
+  async scrollQuery(esIndex, esType, {
+    filter,
+    fields,
+    sort,
+  }) {
+    if (!esIndex || !esType) {
+      throw new CodedError(
+        400,
+        'Invalid es index or es type name',
+      );
+    }
+    const fieldsNotBelong = _.difference(fields, this.getESFields(esIndex).fields);
+    if (fieldsNotBelong.length > 0) {
+      throw new CodedError(
+        400,
+        `Invalid fields: "${fieldsNotBelong.join('", "')}"`,
+      );
+    }
+    const validatedQueryBody = filter ? { query: filter } : {};
+    log.debug('[ES.scrollQuery] scroll query body: ', JSON.stringify(validatedQueryBody, null, 4));
+
+    let currentBatch;
+    let scrollID;
+    let totalData = [];
+    let batchSize = 0;
+
+    while (!currentBatch || batchSize > 0) {
+      if (typeof scrollID === 'undefined') { // first batch
+        const res = await this.client.search({ // eslint-disable-line no-await-in-loop
+          index: esIndex,
+          type: esType,
+          body: validatedQueryBody,
+          scroll: '2m',
```

Comment:
Isn't the default 1 minutes is enough? This timeout is only for ES to process the batch, not the whole results.